public/src/managers/CourseManager.ts

```diff
@@ -332,47 +334,39 @@ export class CourseManager {
     }
 
     /**
-     * Add IStudentSubmissions to an IAssignmentLink
+     * Add lab submissions for every course lab
+     * to the given student or group enrollment.
+     * Used on StudentPage.
      */
-    public async fillLinks(student: User, studentCourse: IStudentLabsForCourse, assignments?: Assignment[]): Promise<void> {
-        if (!studentCourse.enrollment) {
+    public async fillLinks(courseLabLink: IStudentLabsForCourse, student?: User, group?: Group,  assignments?: Assignment[]): Promise<void> {
+        if (!(student || group)) {
             return;
         }
-        if (!assignments) {
-            assignments = await this.getAssignments(studentCourse.course.getId());
-        }
-        if (assignments.length > 0) {
-            const submissions =
-                await this.courseProvider.getAllLabInfos(studentCourse.course.getId(), student.getId());
-
-            for (const a of assignments) {
-                if (!a.getIsgrouplab()) {
-                    const submission = submissions.find((sub) => sub.assignmentid === a.getId());
-                    studentCourse.labs.push({ assignment: a, submission: submission, authorName: student.getName()});
-                }
+
+        if (!assignments || assignments.length < 1) {
+            assignments = await this.getAssignments(courseLabLink.course.getId());
+            if (assignments.length < 1) {
+                return;
             }
         }
-    }
-
-    /**
-     * Add group submissions to an IAssignmentLink
-     */
-    public async fillLinksGroup(group: Group, groupCourse: IStudentLabsForCourse, assignments?: Assignment[]):
-        Promise<void> {
-        if (!groupCourse.enrollment) {
-            return;
+        let submissions : ISubmission[] = [];
+        let labAuthorName = "";
+        if (student) {
+            submissions =
+                await this.courseProvider.getAllLabInfos(courseLabLink.course.getId(), student.getId());
+            labAuthorName = student.getName();
         }
-        if (!assignments) {
-            assignments = await this.getAssignments(groupCourse.course.getId());
+
+        if (group) {
```

Comment:
Should this be?:
```typescript
if (student) {
 ...
} else if (group) {
   submissions = await ....
   ...
} else {
   return;
}
```

Reply:
Well, we check already that it's either or, but can't be too careful. Changed.